addons/mrp_subcontracting/models/stock_picking.py

```diff
@@ -41,47 +41,59 @@ def _compute_display_action_record_components(self):
     # -------------------------------------------------------------------------
     def _action_done(self):
         res = super(StockPicking, self)._action_done()
+
+        for move in self.move_lines.filtered(lambda move: move.is_subcontract):
+            # Auto set qty_producing/lot_producing_id of MO if there isn't tracked component
+            # If there is tracked component, the flow use subcontracting_record_component instead
+            if move._has_tracked_subcontract_components():
+                continue
+            production = move.move_orig_ids.production_id.filtered(lambda p: p.state not in ('done', 'cancel'))[-1:]
+            if not production:
+                continue
+            # Manage additional quantities
+            quantity_done_move = move.product_uom._compute_quantity(move.quantity_done, production.product_uom_id)
+            if float_compare(production.product_qty, quantity_done_move, precision_rounding=production.product_uom_id.rounding) == -1:
+                change_qty = self.env['change.production.qty'].create({
+                    'mo_id': production.id,
+                    'product_qty': quantity_done_move
+                })
+                change_qty.with_context(skip_activity=True).change_prod_qty()
+            # Create backorder MO for each move lines
+            for move_line in move.move_line_ids:
+                # The first MO will have the qty_producing == 0.0
+                if not float_is_zero(production.qty_producing, precision_rounding=production.product_uom_id.rounding):
+                    backorder = production._generate_backorder_productions(close_mo=False)
+                    # The move_dest_ids won't be set because the _split filter out done move
+                    backorder.move_finished_ids.filtered(lambda mo: mo.product_id == move.product_id).move_dest_ids = production.move_finished_ids.filtered(lambda mo: mo.product_id == move.product_id).move_dest_ids
+                    production.product_qty = production.qty_producing
```

Comment:
should be done via _get_quantity_to_backorder() in _generate_backorder_productions()